game/dota_addons/dota_imba/scripts/vscripts/hero/hero_silencer.lua

```diff
@@ -1,181 +1,52 @@
---[[	Author: X-TheDark
-		Date: 10.12.2016	]]
+LinkLuaModifier("modifier_imba_silencer_glaives_hit_counter", "hero/hero_silencer/imba_glaives_of_wisdom", LUA_MODIFIER_MOTION_NONE)
+LinkLuaModifier("modifier_imba_silencer_glaives_int_damage", "hero/hero_silencer/imba_glaives_of_wisdom", LUA_MODIFIER_MOTION_NONE)
 
-function ArcaneCurseCast( keys )
+function GlaiveHit( keys )
 	local caster = keys.caster
 	local target = keys.target
 	local ability = keys.ability
-	local curse_debuff = keys.curse_debuff
-
-	target:AddNewModifier(caster, ability, curse_debuff, nil)
-end
+	local intellect_damage_pct  = ability:GetSpecialValueFor("intellect_damage_pct")
+	local hits_to_silence = ability:GetSpecialValueFor("hits_to_silence")
+	local hit_count_duration = ability:GetSpecialValueFor("hit_count_duration")
+	local silence_duration = ability:GetSpecialValueFor("silence_duration")
+	local int_reduction_pct = ability:GetSpecialValueFor("int_reduction_pct")
+	local int_reduction_duration = ability:GetSpecialValueFor("int_reduction_duration")
+	local caster_int = caster:GetIntellect()
 
-function IntToDamage( keys )
-	local caster = keys.caster
-	local target = keys.target
-	local ability = keys.ability
+	--[[ SILENCER TALENT CHECK - replace hardcoded numbers
```

Comment:
Search for `SILENCER TALENT CHECK` to find all talent checks, I've used this format everywhere